app/main.py

```
@@ -1,25 +1,88 @@
-from abc import ABC
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __get__(self, instance: object, owner: type[object]) -> None:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if self.min_amount <= value <= self.max_amount:
+            setattr(instance, self.protected_name, value)
+        else:
+            setattr(instance, self.protected_name, None)
```

raise ValueError instead of setting the value to None